web_src/js/features/repo-settings.js

```diff
@@ -90,3 +90,12 @@ export function initRepoSettingBranches() {
     });
   }
 }
+
+export function initRepoSettingMirror() {
+  if ($('#push_mirror_use_ssh').length === 0) return;
+
+  $('#push_mirror_use_ssh').on('change', function () {
+    $('#push_mirror_password').parent().toggleClass('disabled', this.checked);
+    $('#push_mirror_username').parent().toggleClass('disabled', this.checked);
+  });
```

Comment:
```suggestion
function togglePushMirrorUsernameAndPassword(disable) {
  $('#push_mirror_password').parent().toggleClass('disabled', disable);
  $('#push_mirror_username').parent().toggleClass('disabled', disable);
}
export function initRepoSettingMirror() {
  const $SSHForPushMirror = $('#push_mirror_use_ssh');
  if ($SSHForPushMirror.length === 0) return;

  if($SSHForPushMirror.checked) togglePushMirrorUsernameAndPassword(true);
  $SSHForPushMirror.on('change', function () {
    togglePushMirrorUsernameAndPassword(this.checked);
  });
```
Probably something like this should fix the issue I mentioned above.
Because of jQuery, however, I've no idea if that actually works.